File: app/src/main/java/com/mavenkalabs/adskipper/rules/MustExistRule.java
package com.mavenkalabs.adskipper.rules;

import android.view.accessibility.AccessibilityNodeInfo;

import androidx.annotation.NonNull;

import java.util.List;
import java.util.Map;

public class MustExistRule extends  BaseIdRule {
    public MustExistRule(String id, String packageName) {
        super(id, packageName);
    }

    @Override
    public RuleResult apply(AccessibilityNodeInfo node, Map<String, Object> parameters) {
        List<AccessibilityNodeInfo> nodes = node.findAccessibilityNodeInfosByViewId(this.qualifiedId);
        return new RuleResult(nodes != null && !nodes.isEmpty(), nodes);
    }

    @NonNull
    @Override
    public String toString() {
        return qualifiedId.split("/")[1];
    }
}

File: app/src/main/kotlin/com/mavenkalabs/adskipper/rules/MustExistRule.kt
package com.mavenkalabs.adskipper.rules

import android.view.accessibility.AccessibilityNodeInfo

class MustExistRule(id: String, packageName: String) : BaseIdRule(id, packageName) {
    override fun apply(
        node: AccessibilityNodeInfo,
        parameters: Map<String, Any>?
    ): RuleResult {
        val nodes = node.findAccessibilityNodeInfosByViewId(this.qualifiedId)
        return RuleResult(nodes?.isNotEmpty() == true, nodes)
    }

    override fun toString() = id
}
